pkg/crosscluster/logical/offline_initial_scan_processor.go

```diff
@@ -244,9 +240,16 @@ func (o *offlineInitialScanProcessor) Next() (rowenc.EncDatumRow, *execinfrapb.P
 	}
 
 	select {
-	case resolved, ok := <-o.checkpointCh:
+	case checkpoint, ok := <-o.checkpointCh:
+		if checkpoint.afterInitialScanCompletion {
```

Comment:
nit: when there are a set of mutually exlusive code paths, I think a switch documents the intent more clearly.

```
switch {
   case !ok:
   // fatal
   case checkpoint.afterInitialScanCompletion:
   // drain
   default:
   // handle span update
} 
```

Reply:
Good idea! fixed.